codex/rest/api.nim

```diff
@@ -177,6 +155,68 @@ proc initRestApi*(node: CodexNodeRef, conf: CodexConf): RestRouter =
         if not stream.isNil:
           await stream.close()
 
+proc initSalesApi(node: CodexNodeRef, router: var RestRouter) =
+  router.api(
+    MethodGet,
+    "/api/codex/v1/sales/slots") do () -> RestApiResponse:
+      ## Returns active slots for the host
+
+      without contracts =? node.contracts.host:
+        return RestApiResponse.error(Http503, "Sales unavailable")
+
+      let json = %(await contracts.sales.mySlots())
+      return RestApiResponse.response($json, contentType="application/json")
+
+  router.api(
+    MethodGet,
+    "/api/codex/v1/sales/availability") do () -> RestApiResponse:
+      ## Returns storage that is for sale
+
+      without contracts =? node.contracts.host:
+        return RestApiResponse.error(Http503, "Sales unavailable")
+
+      without avails =? (await contracts.sales.context.reservations.all(Availability)), err:
+        return RestApiResponse.error(Http500, err.msg)
+
+      let json = %avails
+      return RestApiResponse.response($json, contentType="application/json")
+
+  router.rawApi(
+    MethodPost,
+    "/api/codex/v1/sales/availability") do () -> RestApiResponse:
+      ## Add available storage to sell
+      ##
+      ## size           - size of available storage in bytes
+      ## duration       - maximum time the storage should be sold for (in seconds)
+      ## minPrice       - minimum price to be paid (in amount of tokens)
+      ## maxCollateral  - maximum collateral user is willing to pay per filled Slot (in amount of tokens)
+
+      without contracts =? node.contracts.host:
+        return RestApiResponse.error(Http503, "Sales unavailable")
+
+      let body = await request.getBody()
+
+      without restAv =? RestAvailability.fromJson(body), error:
+        return RestApiResponse.error(Http400, error.msg)
+
+      let reservations = contracts.sales.context.reservations
+
+      if not reservations.hasAvailable(restAv.size.truncate(uint)):
+        return RestApiResponse.error(Http422, "Not enough storage quota")
+
+      without availability =? (
+        await reservations.createAvailability(
+          restAv.size,
+          restAv.duration,
+          restAv.minPrice,
+          restAv.maxCollateral)
+        ), error:
+        return RestApiResponse.error(Http500, error.msg)
+
+      return RestApiResponse.response(availability.toJson,
+                                      contentType="application/json")
+
+proc initStorageApi(node: CodexNodeRef, router: var RestRouter) =
```

Comment:
This is the reason I suggest we don't make the changes to split the initialization into different procs, this will only generate noise and make reviewing the changes more cumbersome. @benbierens I bet if we leave this particular set of changes out, it will be much easier to move this forward.

Reply:
I still think it would be better to have the API calls organized in nicely labeled groups. I don't mind renames. Thinking about names and whether things make sense together is good. I'm convinced the effort will pay itself back.
So I'll rename to purchasing API.

Reply:
From this pov we can see that the debug/info is not purely for debugging. Since it is how we can fetch (updated) SPRs, which are a vital part of how we build up the network, and definitely not 'just' debugging functionality. So, it looks to me that there should be a new category, like `Networking API` or something. And it should contain a way to fetch the SPR, and the current `/connect/{peerid}` call.
This is not a change I'm going to include here, or any time soon. :D

Reply:
What Dmitiy was pointing out was, that this is becoming "monster PR" with too many different changes incorporated in one PR. Generally, it is better to have one change = one PR. I am not against the reshuffling but it should have been in separate PR...

Reply:
I understand that. I don't think extracting a few very simple methods from a big one is cause of concern though. I didn't introduce any complexity with this, it's just some basic clean-up.

Reply:
Well, you introduce reviewing complexity and that is what @dryajov was pointing out.

Reply:
Yeah, thats my point. Just doing this in a separate PR to ease the review of an already large PR.